tests/DCPS/Compiler/xcdr/xcdr.cpp

```diff
@@ -52,6 +52,45 @@ void expect_values_equal(const TypeA& a, const TypeB& b)
   expect_values_equal_base(a, b);
 }
 
+template<>
+void expect_values_equal(const LengthCodeStruct& a, const LengthCodeStruct& b)
+{
+  EXPECT_EQ(a.o, b.o);
+  EXPECT_EQ(a.s, b.s);
+  EXPECT_EQ(a.l, b.l);
+  EXPECT_EQ(a.ll, b.ll);
+
+  EXPECT_EQ(a.b3.a, b.b3.a);
+  EXPECT_EQ(a.b3.b, b.b3.b);
+  EXPECT_EQ(a.b3.c, b.b3.c);
+
+  EXPECT_EQ(a.o5.a, b.o5.a);
+  EXPECT_EQ(a.o5.b, b.o5.b);
+  EXPECT_EQ(a.o5.c, b.o5.c);
+  EXPECT_EQ(a.o5.d, b.o5.d);
+  EXPECT_EQ(a.o5.e, b.o5.e);
+
+  EXPECT_EQ(a.s3.x, b.s3.x);
+  EXPECT_EQ(a.s3.y, b.s3.y);
+  EXPECT_EQ(a.s3.z, b.s3.z);
+
+  EXPECT_EQ(a.t7.s3.x, b.t7.s3.x);
+  EXPECT_EQ(a.t7.s3.y, b.t7.s3.y);
+  EXPECT_EQ(a.t7.s3.z, b.t7.s3.z);
+  EXPECT_EQ(a.t7.o, b.t7.o);
+
+  EXPECT_EQ(a.l3.a, b.l3.a);
+  EXPECT_EQ(a.l3.b, b.l3.b);
+  EXPECT_EQ(a.l3.c, b.l3.c);
+/*
+  EXPECT_EQ(a.str1, b.str1);
```

Comment:
```suggestion
  EXPECT_STREQ(a.str1, b.str1);
```

[See here for the macros](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#string-comparison)